jetbrains.plugins: add several plugins

Nindouja opened the pull request:
Some plugins I regularly use and would like to see in the repo:
* [nix-lsp](https://plugins.jetbrains.com/plugin/25594-nix-lsp)
* [cognitivecomplexity](https://plugins.jetbrains.com/plugin/12024-cognitivecomplexity)
* [string-manipulation](https://plugins.jetbrains.com/plugin/2162-string-manipulation)
* [mermaid-chart](https://plugins.jetbrains.com/plugin/23043-mermaid-chart)
* [code-complexity](https://plugins.jetbrains.com/plugin/21667-code-complexity)
* [grep-console](https://plugins.jetbrains.com/plugin/7125-grep-console)
* [key-promoter-x](https://plugins.jetbrains.com/plugin/9792-key-promoter-x)
* [rainbow-brackets](https://plugins.jetbrains.com/plugin/10080-rainbow-brackets)
* [rainbow-csv](https://plugins.jetbrains.com/plugin/12896-rainbow-csv)
* [randomness](https://plugins.jetbrains.com/plugin/9836-randomness)
* [gittoolbox](https://plugins.jetbrains.com/plugin/7499-gittoolbox)
* [wakatime](https://plugins.jetbrains.com/plugin/7425-wakatime)
* [extra-ide-tweaks](https://plugins.jetbrains.com/plugin/23927-extra-ide-tweaks)
* [extra-toolwindow-colorful-icons](https://plugins.jetbrains.com/plugin/16604-extra-toolwindow-colorful-icons)
* [extra-icons](https://plugins.jetbrains.com/plugin/11058-extra-icons)
* [extra-tools-pack](https://plugins.jetbrains.com/plugin/24559-extra-tools-pack)
* [mario-progress-bar](https://plugins.jetbrains.com/plugin/14708-mario-progress-bar)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

FliegendeWurst reviewed (commented):
Would be good to squash the commits. Please remove the merge commit and rebase instead

Nindouja commented:
> Would be good to squash the commits. Please remove the merge commit and rebase instead

Everything should be clean. I also added [one other plugin](https://plugins.jetbrains.com/plugin/6884-handlebars-mustache) in the process

leona-ya reviewed (approved):
Thanks! Sorry that it took so long. I'm currently working through the jetbrains backlog.

leona-ya commented:
Can you rebase this, then I will merge.

Nindouja commented:
Thanks for the review @leona-ya. The rebase was kinda messy so I prefered starting fresh from master. As all my changes were made via the update_plugin script is way less work, and should be much more cleaner.

I've added also added a few plugins, here is the full list:

- https://plugins.jetbrains.com/plugin/8607-nixidea
- https://plugins.jetbrains.com/plugin/17718-github-copilot
- https://plugins.jetbrains.com/plugin/7793-markdown
- https://plugins.jetbrains.com/plugin/20146-mermaid
- https://plugins.jetbrains.com/plugin/10037-csv-editor
- https://plugins.jetbrains.com/plugin/25594-nix-lsp
- https://plugins.jetbrains.com/plugin/12024-cognitivecomplexity
- https://plugins.jetbrains.com/plugin/2162-string-manipulation
- https://plugins.jetbrains.com/plugin/23043-mermaid-chart
- https://plugins.jetbrains.com/plugin/21667-code-complexity
- https://plugins.jetbrains.com/plugin/7125-grep-console
- https://plugins.jetbrains.com/plugin/9792-key-promoter-x
- https://plugins.jetbrains.com/plugin/10080-rainbow-brackets
- https://plugins.jetbrains.com/plugin/12896-rainbow-csv
- https://plugins.jetbrains.com/plugin/9836-randomness
- https://plugins.jetbrains.com/plugin/7499-gittoolbox
- https://plugins.jetbrains.com/plugin/7425-wakatime
- https://plugins.jetbrains.com/plugin/23927-extra-ide-tweaks
- https://plugins.jetbrains.com/plugin/16604-extra-toolwindow-colorful-icons
- https://plugins.jetbrains.com/plugin/11058-extra-icons
- https://plugins.jetbrains.com/plugin/24559-extra-tools-pack
- https://plugins.jetbrains.com/plugin/14708-mario-progress-bar
- https://plugins.jetbrains.com/plugin/21551-ferris
- https://plugins.jetbrains.com/plugin/6884-handlebars-mustache
- https://plugins.jetbrains.com/plugin/7177-file-watchers
- https://plugins.jetbrains.com/plugin/7275-codeglance
- https://plugins.jetbrains.com/plugin/13308-indent-rainbow
- https://plugins.jetbrains.com/plugin/18824-codeglance-pro
- https://plugins.jetbrains.com/plugin/22857-gitlab
- https://plugins.jetbrains.com/plugin/8195-toml
- https://plugins.jetbrains.com/plugin/21962-dev-containers
- https://plugins.jetbrains.com/plugin/4230-bashsupport
- https://plugins.jetbrains.com/plugin/9525--env-files-support
- https://plugins.jetbrains.com/plugin/18922-gerry-themes
- https://plugins.jetbrains.com/plugin/26084-markdtask
- https://plugins.jetbrains.com/plugin/164-ideavim
- https://plugins.jetbrains.com/plugin/21904-developer-tools
- https://plugins.jetbrains.com/plugin/14823-full-line-code-completion

Nindouja commented:
@leona-ya would you have any time to take a look at the latest commits and merge that ?

FliegendeWurst commented on pkgs/applications/editors/jetbrains/plugins/plugins.json:
```diff
@@ -185,6 +309,37 @@
+        "rust-rover",
+        "webstorm"
+      ],
+      "builds": {
+        "241.19072.1155": null,
+        "243.22562.112": null,
+        "243.22562.115": null,
+        "243.22562.145": null,
+        "243.22562.151": null,
+        "243.22562.155": null,
+        "243.22562.164": null,
+        "243.22562.171": null,
+        "243.22562.180": null,
+        "243.22562.186": null,
+        "243.22562.187": null
+      },
```
Is this supposed to be all nulls?

Nindouja replied:
Well I don't really knows because I made all changes via the Python script, I did not edit the file manually. Also when there was a rebase in order to not mess up anything I started fresh from master and add all plugins again via the script only.
From my understanfding @leona-ya is familiar with that part way more than I'm and maybe thy can answer your question better.

FliegendeWurst replied:
I also had a look, it seems that the plugin is unusable without at least one non-null value.

leona-ya replied:
I think this happens when there is no compatible version with any of the IDE versions we currently support. So I'd vote to drop this plugin.

Nindouja commented:
I've reset everything one more time. I took care to remove any plugin with zero compatibility. This should be all good now.

leona-ya commented:
Could you squash the commits together please? Then I should be able to merge it. Thanks

Nindouja committed:
Adding multiple IDEA plugins

github-copilot
continue
dot-language
better-direnv
ansi-highlighter-premium
nixidea
mermaid
csv-editor
nix-lsp
cognitivecomplexity
string-manipulation
mermaid-chart
code-complexity
grep-console
key-promoter-x
rainbow-brackets
rainbow-csv
randomness
gittoolbox
wakatime
extra-ide-tweaks
extra-toolwindow-colorful-icons
extra-icons
extra-tools-pack
mario-progress-bar
ferris
handlebars-mustache
file-watchers
indent-rainbow
codeglance-pro
gitlab
toml
dev-containers
env-files-support
gerry-themes
markdtask
ideavim
developer-tools

Nindouja commented:
@leona-ya thanks for the time. I've squashed and added the list.

leona-ya reviewed (approved):
Thank you for your contribution and your patience.

JuliusFreudenberger commented:
@leona-ya is it possible to backport this or should a backport of adding plugins not be done?
It breaks the automatic backport, e.g. in https://github.com/NixOS/nixpkgs/pull/381402#issuecomment-2660952630
I stumbled across the commit introduced with this PR while trying to perform a manual backport.

FliegendeWurst commented:
New packages are usually safe to backport.

dotlambda reviewed (commented):
Commit message doesn't follow guidelines.

JuliusFreudenberger commented:
> Commit message doesn't follow guidelines.

@dotlambda, this PR is merged for over a month.
